bundles/async_remote_services/admin/src/admin.cc

```diff
@@ -21,300 +21,256 @@
 }
 
-void celix::async_rsa::AsyncAdmin::addService(void *svc, const celix_properties_t *props) {
-    auto *objectClass = celix_properties_get(props, OSGI_FRAMEWORK_OBJECTCLASS, nullptr);
-    auto *remote = celix_properties_get(props, "remote", nullptr);
-    auto endpointId = celix_properties_get(props, ENDPOINT_IDENTIFIER, nullptr);
+void celix::async_rsa::AsyncAdmin::addService(const std::shared_ptr<void>& svc, const std::shared_ptr<const celix::Properties>& props) {
+    assert(props->getAsBool(celix::rsa::REMOTE_SERVICE_PROPERTY_NAME, false)); //"only remote services should be tracker by the service tracker");
 
-    if(objectClass == nullptr) {
+    auto serviceName = props->get(celix::SERVICE_NAME, "");
+    if (serviceName.empty()) {
         L_WARN("Adding service to be exported but missing objectclass");
         return;
     }
 
-    if(remote == nullptr) {
-        // not every service is remote, this is fine but not something the RSA admin is interested in.
-        return;
-    } else {
-        L_WARN("found remote service %s", objectClass);
-    }
-
-    if(endpointId == nullptr) {
-        L_WARN("Adding service to be exported but missing endpoint.id");
-        return;
-    }
-
-    std::unique_lock l(_m);
-    auto factory = _exportedServiceFactories.find(objectClass);
-
-    if(factory == end(_exportedServiceFactories)) {
-        L_WARN("Adding service to be exported but no factory available yet, delaying creation");
-
-        Properties cppProps;
-
-        auto it = celix_propertiesIterator_construct(props);
-        const char* key;
-        while(key = celix_propertiesIterator_nextKey(&it), key != nullptr) {
-            cppProps.set(key, celix_properties_get(props, key, nullptr));
-        }
-
-        _toBeCreatedExportedEndpoints.emplace_back(std::make_pair(svc, std::move(cppProps)));
-        return;
-    }
-
-    _exportedServiceInstances.emplace(endpointId, factory->second->create(svc, endpointId));
+    std::lock_guard<std::mutex> lock{_m};
+    _toBeExportedServices.emplace_back(svc, props);
+    createExportServices();
 }
 
-void celix::async_rsa::AsyncAdmin::removeService(void *, const celix_properties_t *props) {
-    auto *objectClass = celix_properties_get(props, OSGI_FRAMEWORK_OBJECTCLASS, nullptr);
-    auto *remote = celix_properties_get(props, "remote", nullptr);
-    auto svcId = celix_properties_get(props, ENDPOINT_IDENTIFIER, nullptr);
-
-    if(objectClass == nullptr) {
-        L_WARN("Removing service to be exported but missing objectclass");
-        return;
-    }
+void celix::async_rsa::AsyncAdmin::removeService(const std::shared_ptr<void>& /*svc*/, const std::shared_ptr<const celix::Properties>& props) {
+    assert(props->getAsBool(celix::rsa::REMOTE_SERVICE_PROPERTY_NAME, false)); //"only remote services should be tracker by the service tracker");
```

Comment:
`REMOTE_SERVICE_PROPERTY_NAME` implies a string, if it is a string, `get` is probably a better match than `getAsBool`

Reply:
done